transport/src/main/java/io/netty/channel/AbstractChannelHandlerContext.java

```diff
@@ -153,7 +150,10 @@ public ChannelHandlerContext fireUserEventTriggered(Object event) {
     @Override
     public ChannelHandlerContext fireChannelRead(Object msg) {
         AbstractChannelHandlerContext next = findContextInbound();
-        ReferenceCountUtil.touch(msg, next);
+        if (ResourceLeakDetector.ENABLED) {
```

Comment:
@ninja- @Scottmitch actually I'm really not sure we gain anything as `ResourceLeakDetector.record(...)` will only ever called if the leak detection is enabled. Otherwise this is really just a "NO OP".